base/linalg/blas.jl

```diff
@@ -112,6 +112,29 @@ function set_num_threads(n::Integer)
     return nothing
 end
 
+"""
+    get_num_threads(n)
+
+Get the number of threads the BLAS library can use.
+"""
+function get_num_threads()
```

Comment:
Try-catches can be a bit expensive but I'm not sure what better approaches there might be. `Libdl.dlsym` might be a faster path?